nuxeo-services/login/nuxeo-platform-login-openid/src/test/java/org/nuxeo/ecm/platform/oauth2/openid/TestProvider.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * (C) Copyright 2013 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.nuxeo.ecm.platform.oauth2.openid;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.ecm.core.test.annotations.Granularity;
+import org.nuxeo.ecm.core.test.annotations.RepositoryConfig;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+@RunWith(FeaturesRunner.class)
+@Features(CoreFeature.class)
+@RepositoryConfig(cleanup = Granularity.METHOD)
+@Deploy("org.nuxeo.ecm.directory.sql")
+@Deploy("org.nuxeo.ecm.directory")
+@Deploy("org.nuxeo.ecm.directory.api")
+@Deploy("org.nuxeo.ecm.platform.login.openid.test")
+@Deploy("org.nuxeo.ecm.platform.login.openid.test:OSGI-INF/openid-provider-contrib.xml")
+public class TestProvider {
+
+    @Test
+    public void verifyProviderRegistration() {
+
+        OpenIDConnectProviderRegistry registry = Framework.getService(OpenIDConnectProviderRegistry.class);
+        Assert.assertNotNull(registry);
+
+        OpenIDConnectProvider provider = registry.getProvider("MY_NAME");
+        Assert.assertNotNull(provider);
+        Assert.assertEquals("MY_AUTHENTICATION_METHOD", provider.authenticationMethod);
+
+    }
+
+}
```

Comment:
Knowing that the existing `TestOpenIDProviders` already tests the provider registration, it sounds overkill to add an extra test class just to validate this new element.
WDYT of just adding:
```
        // check provider's authenticationMethod
        provider = registry.getProvider("MY_NAME");
        Assert.assertNotNull(provider);
        Assert.assertEquals("MY_AUTHENTICATION_METHOD", provider.authenticationMethod);
```
to `TestOpenIDProviders`, and:
```
    <provider>
      <name>MY_NAME</name>              
      <authenticationMethod>MY_AUTHENTICATION_METHOD</authenticationMethod>
    </provider>
```
to `openid-google-contrib.xml` (minimal contribution here to validate the new element)?

Then `openid-provider-contrib.xml` wouldn't be needed anymore.

Reply:
Bonus point (optional, really): you could clean up the test by removing some useless code and follow the usual practice:
- static imports for junit assert methods
- minimal test feature needed, here `RuntimeFeature`
- no useless `@Deploy` and `@RepositoryConfig` statements
- use of service injection
```
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

import javax.inject.Inject;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.nuxeo.runtime.test.runner.Deploy;
import org.nuxeo.runtime.test.runner.Features;
import org.nuxeo.runtime.test.runner.FeaturesRunner;
import org.nuxeo.runtime.test.runner.RuntimeFeature;

@RunWith(FeaturesRunner.class)
@Features(RuntimeFeature.class)
@Deploy("org.nuxeo.ecm.platform.login.openid.test")
public class TestOpenIDProviders {

    @Inject
    protected OpenIDConnectProviderRegistry registry;

    @Test
    public void verifyServiceRegistration() {

        OpenIDConnectProvider provider = registry.getProvider("TestingGoogleOpenIDConnect");
        assertNotNull(provider);

        assertTrue(!OpenIDConnectHelper.getEnabledProviders().isEmpty());

        OpenIDConnectProvider provider2 = registry.getProvider("TestingGoogleOpenIDConnect2");
        assertNotNull(provider2);

        // check provider's authenticationMethod
        provider = registry.getProvider("MY_NAME");
        assertNotNull(provider);
        assertEquals("MY_AUTHENTICATION_METHOD", provider.authenticationMethod);
    }

}
```

Reply:
Done